packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs

```diff
@@ -559,6 +559,8 @@ private void AppendType(CSharpType type, bool isDeclaration, bool writeTypeNameO
                 AppendRaw("global::");
                 AppendRaw(type.Namespace);
                 AppendRaw(".");
+                if (type.DeclaringType is not null)
+                    AppendRaw($"{type.DeclaringType.Name}.");
                 AppendRaw(type.Name);
```

Comment:
would it be more reasonable if write this in this way:
```csharp
if (type.DeclaringType is {} declaringType)
{
	writeTypeName(declaringType);
}
else
{
	AppendRaw("global::");
        AppendRaw(type.Namespace);
        AppendRaw(".");
}
AppendRaw(".");
AppendRaw(type.Name);
```

Reply:
what does `writeTypeName` do here?